### ISSUE TITLE
Improve `performance-move-const-arg` message when no move constructor is available

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#62550]
```c++
#include <string>
#include <utility>

struct S {
    S();
    S(const S&);

    std::string s;
};

extern void cb1(S);
extern void cb2(const std::string&);

void f()
{
    S s;
    cb1(std::move(s));

    std::string str;
    cb2(std::move(str));
}
```

```console
<source>:17:9: warning: passing result of std::move() as a const reference argument; no move will actually happen [performance-move-const-arg]
    cb1(std::move(s));
        ^~~~~~~~~~ ~
<source>:20:9: warning: passing result of std::move() as a const reference argument; no move will actually happen [performance-move-const-arg]
    cb2(std::move(str));
        ^~~~~~~~~~   ~
```
https://godbolt.org/z/Ga3hM5h6K

Both cases trigger the exactly same warning but for different reasons. It would be great if the first case could point out that no move constructor is available/accessible instead. It already provides more details e.g. if you try to move a trivally-copyable object, so the check already provides different messages.

The original code has a much bigger class and it took me some reducing to actually figure out what the issue is as the issue was not being triggered in all cases (still investigating that).

**Comment by AMS21** [comment]
https://reviews.llvm.org/D153220
